src/kubeobjects/query/dynakube.go

```diff
@@ -0,0 +1,83 @@
+package query
+
+import (
+	"context"
+
+	dynatracev1beta1 "github.com/Dynatrace/dynatrace-operator/src/api/v1beta1"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	trustedCAKey = "certs"
+	proxyKey     = "proxy"
+	tlsCertKey   = "server.crt"
+)
+
+type DynakubeQuery struct {
+	clt       client.Client
+	namespace string
+}
+
+func NewDynakubeQuery(clt client.Client, namespace string) *DynakubeQuery {
```

Comment:
Queries shouldn't mutate object, so value should be returned instead of a pointer and used as method receivers.